### ISSUE TITLE
Keyboard input

### ISSUE THREAD

**Issue by rogalian** [MiSTer-devel/Arcade-Galaxian_MiSTer#58]
Is the mapping shown in the readme still valid? I've tried multiple keyboards but I'm getting no response from the keys shown.

-- Keyboard inputs :
--
--   F2         : Coin + Start 2 players
--   F1         : Coin + Start 1 player
--   SPACE,CTRL : Fire
--   LEFT,RIGHT : Movements
--
-- MAME/IPAC/JPAC Style Keyboard inputs:
--   5           : Coin 1
--   6           : Coin 2
--   1           : Start 1 Player
--   2           : Start 2 Players
--   D,G         : Player 2 Movements
--   A           : Player 2 Fire

**Comment by sorgelig** [comment]
for keyboards input It's outdated.
MiSTer supports buttons defined on keyboard. So just define keyboard keys as you wish.

IPAC/JPAC should be valid if input device is properly identified as such.

**Comment by rogalian** [comment]
Fair enough, readme out of date. Shame, I've been unable to get IPAC/JPAC method working either, the only thing I've managed is to define keyboard mappings, which is limited as Player 2 inputs are not presented for mapping.

(I'm using an IrkenLabs JAMMA expander in a JAMMA modified original Galaxian cocktail cabinet, with independent P1 and P2 sides)

**Comment by sorgelig** [comment]
Check MiSTer.ini - there you will find options to define IPAC/JPAC

**Comment by rogalian** [comment]
I will try those again, but I couldn't get them going yesterday.

This is specific to when the core is being run with the Cocktail dipswitch enabled. There are two sets of controls, unlike the upright cabinet.

**Comment by rogalian** [comment]
Okay, I'll explain this differently.

In cocktail mode, player 1 controls should not control player 2 ship. For upright cabinets the current control methods work fine, but in cocktail there are distinct left/right/fire for player 2 as shown in the out-of-date readme.

Ignoring the JAMMA expander, it is not possible to use any USB keyboard for this behaviour - player 1 controls are used for player 2 which is incorrect.

**Comment by sorgelig** [comment]
Separate controls for player 1 and 2 are used only in arcades where simultaneous of both players are supported. 
Originally MiSTer is made for gamepads, and in arcades where players play separately, you can use the same gamepad.

**Comment by rogalian** [comment]
Galaxian and many other cocktail titles supporting alternate players with screen flipping have distinct controls for P1 and P2 used independently. Arkanoid is another, Galaga etc

Exposing player 2 inputs for use on separate keys of the same keyboard would let the cores be used as a PCB substitute. Niche case but more accurate.
